services/rpcfilters/api.go

```diff
@@ -114,6 +117,51 @@ func (api *PublicAPI) NewPendingTransactionFilter() rpc.ID {
 
 }
 
+// NewFilter creates a new filter and returns the filter id. It can be
+// used to retrieve logs when the state changes. This method cannot be
+// used to fetch logs that are already stored in the state.
+//
+// Default criteria for the from and to block are "latest".
+// Using "latest" as block number will return logs for mined blocks.
+// Using "pending" as block number returns logs for not yet mined (pending) blocks.
+// In case logs are removed (chain reorg) previously returned logs are returned
+// again but with the removed property set to true.
+//
+// In case "fromBlock" > "toBlock" an error is returned.
+//
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_newfilter
+func (api *PublicAPI) NewFilter(crit FilterCriteria) (rpc.ID, error) {
+	api.filtersMu.Lock()
+	defer api.filtersMu.Unlock()
+
+	f := newFilter()
+	id := rpc.ID(uuid.New())
+
+	api.filters[id] = f
+
+	logFilter, err := NewLogFilter(api.blockProvider, &crit)
+
+	go func() {
+		id, s := logFilter.Subscribe()
+		defer logFilter.Unsubscribe(id)
+
+		for {
+			select {
+			case hash := <-s:
```

Comment:
This filter is supposed to return logs (or a list of logs) instead of hashes like the other types of filters. Still trying to figure out what Logs struct will be the right one to use once I understand how to get logs from lists of blocks (and filter criteria)

Reply:
There is this [`eth_getFilterChanges`](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getfilterchanges) specification. Is pretty specific on what exactly should be returned by this filter.